### PR TITLE
put pubkey and ipns entry to dht in parallel

### PR DESCRIPTION
**Pull request by whyrusleeping**
its uh... faster.

License: MIT
Signed-off-by: Jeromy <jeromyj@gmail.com>

### CODE REVIEW

File: namesys/publisher.go [hunk 1]
```diff
@@ -159,6 +169,19 @@ func PutRecordToRouting(ctx context.Context, k ci.PrivKey, value path.Path, seqn
 	return nil
 }
 
+func waitOnErrChan(ctx context.Context, errs chan error) error {
+	select {
+	case err := <-errs:
+		if err != nil {
+			return err
+		}
```
Comment (noffle) [inline namesys/publisher.go]: This `if` can be simplified to just `return err`, right? Then you can drop the final `return nil` too.

**Comment by noffle** [comment]
Minor nit above, but :horse: LGTM.

File: namesys/publisher.go (hunk 1 above)
Comment (whyrusleeping) [inline namesys/publisher.go]: oh, duh. I had it up in the previous function before i broke it out

**Comment by jbenet** [comment]
LGTM too. CI is unhappy.

**Comment by whyrusleeping** [comment]
CI didnt like that there was a race condition on the datastores lol. fixed now.

**Pull request merged**